core/src/main/java/org/infinispan/transaction/impl/RemoteTransaction.java

```diff
@@ -153,4 +159,12 @@ public final TotalOrderRemoteTransactionState getTransactionState() {
          return transactionState;
       }
    }
+
+   public final CompletableFuture<Void> enterSynchronizationAsync(CompletableFuture<Void> releaseFuture) {
+      while (true) {
```

Comment:
I'm pretty sure the JIT implements `while (true)` as an unconditional jump, but I guess `do/while` looks a bit better.